### ISSUE TITLE
Does not preserve comments

### ISSUE THREAD

**Issue by stilvoid** [awslabs/aws-cloudformation-template-formatter#9]
This is a thorny one and, unless I can find one, I'm gonna end up having to write a YAML parser!

**Comment by kadrach** [comment]
`gopkg.in/yaml.v3` appears to support custom tags and comments.

**Comment by stilvoid** [comment]
Yes, I have been looking into this and will try to get it in to a future release :)

**Comment by yingw787** [comment]
I've been experiencing this concern to a minor degree, definitely like YAML over JSON for its comments as inline documentation :) Definitely interested in that future release!

**Comment by edenhare** [comment]
I just looked at cfn-format, and I like it.  It desperately needs to preserve the comments.

**Comment by stilvoid** [comment]
> I just looked at cfn-format, and I like it. It desperately needs to preserve the comments.

100% - I'm working on it - it's quite a large refactor but I anticipate getting it done within the next few weeks

**Comment by edenhare** [comment]
Awesome.  I like it a lot.  I am a member of the AWS Community Builder program, and I write a lot of articles on [Medium](https://labrlearning.medium.com/) about AWS.  I will happily write an article about this once you are done. (Might I also suggest the -w makes a backup copy as well?). If I knew Go, I would help.  If I can beta test, write docs, examples, etc., let me know.

**Comment by xmiao2** [comment]
+1, also like this a lot. I'm trying to add license file header to my yaml templates but the formatter removes them. I had to write a separate script to prefix license headers after running cfn-format.

**Comment by cacack** [comment]
@stilvoid Any luck with this?  I too would like to implement this in our development workflow but not preserving comments is a deal breaker for us..
